external-crates/move/crates/move-compiler/tests/move_2024/typing/index_invalid_crash.exp

```diff
@@ -0,0 +1,12 @@
+error[E04007]: incompatible types
+   ┌─ tests/move_2024/typing/index_invalid_crash.move:14:5
+   │
+12 │ public fun index_by_reference(table: &mut Table<u64, vector<u64>>) {
```

Comment:
This error message looks weird... Shouldn't it be pointing to the `_k` parameter of the `borrow` function?

Reply:
This is a strange error, but since `_k : K`, this really is the concrete type expected, and where it is defined.

Reply:
This is to do with the generics around the test function. Consider this bode:

```
module a::bad_type;

public fun f<T: drop>(x: T, y: T): T {
    x
}

public fun test(x: u64, y: u32): u64 {
    f(x, y)
}
```

Today we give the following error:

```
error[E04007]: incompatible types
  ┌─ ./sources/bad_type.move:8:5
  │
7 │ public fun test(x: u64, y: u32): u64 {
  │                    ---     --- Given: 'u32'
  │                    │
  │                    Expected: 'u64'
8 │     f(x, y)
  │     ^^^^^^^ Invalid call of 'a::bad_type::f'. Invalid argument for parameter 'y'
  ```
  
  This could be shored up in the future, but I'd rather not block this diff on generally improving polymorphic call errors.

Reply:
NP. I just wanted to point it out. The main weirdness here is that this error mentions `_k` with no `_k` in sight in the rest of the error message